### PR TITLE
Fix a bug in custom data split.

### PR DESCRIPTION
**Pull request by zheng-da**
*Description of changes:*
The current data split doesn't work if the input node data are stored in multiple files.
https://github.com/awslabs/graphstorm/issues/297

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by zhjwy9343 — APPROVED** [review]
LGTM

File: python/graphstorm/gconstruct/transform.py [hunk 1]
```diff
@@ -853,13 +855,14 @@ class CustomLabelProcessor:
...
-    def __init__(self, col_name, label_name, task_type,
+    def __init__(self, col_name, label_name, id_col, task_type,
                  train_idx=None, val_idx=None, test_idx=None):
+        self._id_col = id_col
         self._col_name = col_name
         self._label_name = label_name
-        self._train_idx = train_idx
-        self._val_idx = val_idx
-        self._test_idx = test_idx
+        self._train_idx = set(train_idx.tolist()) if train_idx is not None else None
```
Comment (classicsong) [inline python/graphstorm/gconstruct/transform.py:863]: This is a list of strings?

File: python/graphstorm/gconstruct/transform.py [hunk 2]
```diff
@@ -1113,8 +1113,9 @@ def parse_label_ops(confs, is_node):
         val_idx = read_index_json(custom_split['valid']) if 'valid' in custom_split else None
         test_idx = read_index_json(custom_split['test']) if 'test' in custom_split else None
         label_col = label_conf['label_col'] if 'label_col' in label_conf else None
-        return [CustomLabelProcessor(label_col, label_col, task_type,
-                                     train_idx, val_idx, test_idx)]
+        assert "node_id_col" in confs, "Custom data split only works for nodes."
```
Comment (classicsong) [inline python/graphstorm/gconstruct/transform.py:1116]: We need to update the README to introduce "node_id_col".

File: python/graphstorm/gconstruct/transform.py [hunk 3]
```diff
@@ -874,28 +877,30 @@ def label_name(self):
         """
         return self._label_name
 
-    def data_split(self, num_samples):
+    def data_split(self, ids):
```
Comment (classicsong) [inline python/graphstorm/gconstruct/transform.py:880]: Can we add a unit test for CustomLabelProcessor?
We only have end2end test now.

File: python/graphstorm/gconstruct/transform.py (hunk 1 above)
Comment (zheng-da) [reply python/graphstorm/gconstruct/transform.py:863]: these are node IDs. they can be strings or integers.

File: python/graphstorm/gconstruct/transform.py (hunk 2 above)
Comment (zheng-da) [reply python/graphstorm/gconstruct/transform.py:1116]: what do you mean? we have node_id_col all the time

File: python/graphstorm/gconstruct/transform.py (hunk 3 above)
Comment (zheng-da) [reply python/graphstorm/gconstruct/transform.py:880]: we have unit test for custom data split.

Comment (zheng-da) [reply python/graphstorm/gconstruct/transform.py:880]: https://github.com/awslabs/graphstorm/blob/main/tests/unit-tests/gconstruct/test_construct_graph.py#L599-L697

Comment (zheng-da) [reply python/graphstorm/gconstruct/transform.py:880]: we have many unit tests for custom data split.

**Review by classicsong — APPROVED** [review]

**Pull request merged**